### PR TITLE
New product price display and product summary

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Integrate the new display of form price in the form of an addition between `priceTaxExcluded + taxRate = priceTaxIncluded` The product form now includes a product summary which updates automatically when inputs are modifed
| Type?             | new feature
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | ~
| Related PRs       | ~
| How to test?      | This is only a partial feature, the ecotax is still missing so no QA for this one Only green CI The functional tests will be performed on the next PR that integrates ecotax
| Possible impacts? | ~

Capture d’écran 2022-05-10 à 04 26 23

### CODE REVIEW

File: src/PrestaShopBundle/Form/Admin/Sell/Product/Pricing/PricingType.php [hunk 1]
```diff
@@ -46,55 +44,18 @@
...
-     */
     public function __construct(
         TranslatorInterface $translator,
         array $locales,
-        array $taxRuleGroupChoices,
-        array $taxRuleGroupChoicesAttributes,
-        Currency $defaultCurrency,
-        RouterInterface $router,
-        bool $taxEnabled
+        Currency $defaultCurrency
```
Comment (Progi1984) [inline src/PrestaShopBundle/Form/Admin/Sell/Product/Pricing/PricingType.php:55]: Please define BC Breaks

File: src/PrestaShopBundle/Form/Admin/Sell/Product/Pricing/RetailPriceType.php [hunk 2]
```diff
@@ -52,28 +54,49 @@ class RetailPriceType extends TranslatorAwareType
...
+
     public function __construct(
         TranslatorInterface $translator,
         array $locales,
         Locale $contextLocale,
         Currency $defaultCurrency,
+        array $taxRuleGroupChoices,
+        array $taxRuleGroupChoicesAttributes,
+        RouterInterface $router,
+        bool $taxEnabled,
```
Comment (Progi1984) [inline src/PrestaShopBundle/Form/Admin/Sell/Product/Pricing/RetailPriceType.php:90]: Please define BC Breaks

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

File: admin-dev/themes/new-theme/js/components/form/form-object-mapper.ts [hunk 3]
```diff
@@ -451,15 +460,21 @@ export default class FormObjectMapper {
    * This method is called when this component initializes or when triggered by an external event.
    */
   private updateFullObject():void {
-    const serializedForm = this.$form.serializeJSON({
-      checkboxUncheckedValue: '0',
+    // Temporarily enable all inputs or they will not be serialized
```
Comment (matks) [inline admin-dev/themes/new-theme/js/components/form/form-object-mapper.ts:463]: It's a dangerous trick 😅

File: src/PrestaShopBundle/Form/Admin/Sell/Product/Pricing/PricingType.php (hunk 1 above)
Comment (jolelievre) [reply src/PrestaShopBundle/Form/Admin/Sell/Product/Pricing/PricingType.php:55]: No BC break on experimental code

File: src/PrestaShopBundle/Form/Admin/Sell/Product/Pricing/RetailPriceType.php (hunk 2 above)
Comment (jolelievre) [reply src/PrestaShopBundle/Form/Admin/Sell/Product/Pricing/RetailPriceType.php:90]: No BC break on experimental code

**Comment by jolelievre** [comment]
HI @Progi1984 

no need to define BC breaks on this code, it's experimental code for the product page, the decision was shared with @eternoendless and @PierreRambaud There's no interest documenting a bunch of BC break on a code that was never release for production

File: src/Adapter/Session/Repository/CustomerSessionRepository.php [hunk 4]
```diff
@@ -141,7 +141,7 @@ public function clearOutdatedSessions(): void
 
             $qb = $this->connection->createQueryBuilder();
             $qb->delete($this->dbPrefix . 'customer_session')
-                ->where('date_upd < :dateUpdated')
+                ->where('date_upd <= :dateUpdated')
```
Comment (matks) [inline src/Adapter/Session/Repository/CustomerSessionRepository.php:144]: Is this related to this PR?

File: src/Adapter/Session/Repository/EmployeeSessionRepository.php [hunk 5]
```diff
@@ -141,8 +141,8 @@ public function clearOutdatedSessions(): void
 
             $qb = $this->connection->createQueryBuilder();
             $qb->delete($this->dbPrefix . 'employee_session')
-                ->where('date_upd < :dateUpdated')
-                ->setParameter('dateUpdated', $date->format('Y-m-d'));
+                ->where('date_upd <= :dateUpdated')
+                ->setParameter('dateUpdated', $date->format('Y-m-d H:i:s'));
```
Comment (matks) [inline src/Adapter/Session/Repository/EmployeeSessionRepository.php:145]: Same?

**Review by matks — APPROVED** [review]
Nothing blocking 👍

File: src/Adapter/Session/Repository/CustomerSessionRepository.php (hunk 4 above)
Comment (jolelievre) [reply src/Adapter/Session/Repository/CustomerSessionRepository.php:144]: No but as I explained on slack the behat tests were failing I already encountered this bug a few weeks ago, I have the feeling that it happened more at night when there more resources available for CI and they can run faster
My guess is that the test used to pass by chance because enough time had passed to interprate the expiration, bit if it runs fast the values were equal, the condition used AND the data in the scenario were too close to be correctly interprated

File: admin-dev/themes/new-theme/js/components/form/form-object-mapper.ts (hunk 3 above)
Comment (jolelievre) [reply admin-dev/themes/new-theme/js/components/form/form-object-mapper.ts:463]: Not that much because we use exactly the same found input to disable and enable them, so we are sure that everything that was enabled will be disabled again

The risk would be on element which dynamically change the disable mode, but I tested it with some of our disabled elements in the product page that are disabled via a switch and even those work fine

**Review by NeOMakinG — APPROVED** [review]
